src/deferred.js

```diff
@@ -182,6 +187,12 @@ jQuery.extend( {
 							if ( depth ) {
 								process();
 							} else {
+
+								// Call an optional hook to record the stack, in case of exception
+								// since it's otherwise lost when execution goes async
+								if ( jQuery.Deferred.getStackHook ) {
+									process.stackTrace = jQuery.Deferred.getStackHook();
```

Comment:
I'm thinking of something akin to a verbose mode, which could be turned on and run for all deferreds during debugging.